python/ray/dag/compiled_dag_node.py

```diff
@@ -271,7 +277,9 @@ def resolve(self, channel_results: Any) -> Any:
         """
 
         if isinstance(self.input_variant, ChannelInterface):
-            value = channel_results[self.channel_idx]
+            future = channel_results[self.channel_idx]
+            assert isinstance(future, DAGOperationFuture)
+            value = future.wait()
```

Comment:
It seems like it would avoid a lot of the code change if you put the call to `wait` in `reset_intermediate_buffer`. Then we can keep the future concept local to this class and I believe we wouldn't need ListFuture, WrappedFuture, etc?

Reply:
Yeah I think whether to use future or the value in the buffer is not the main issue -- that part is easy. See https://github.com/ray-project/ray/pull/47586#issuecomment-2408292570

Reply:
Done.